packages/rsocket-core/src/RSocketMachine.js

```diff
@@ -892,13 +891,14 @@ class RSocketMachineImpl<D, M> implements RSocketMachine<D, M> {
       flags |= FLAGS.COMPLETE;
       this._subscriptions.delete(streamId);
     }
+    let metadata;
     if (payload.metadata !== undefined &&
         payload.metadata !== null) {
       // eslint-disable-next-line no-bitwise
       flags |= FLAGS.METADATA;
+      metadata = this._serializers.metadata.serialize(payload.metadata);
     }
     const data = this._serializers.data.serialize(payload.data);
-    const metadata = this._serializers.metadata.serialize(payload.metadata);
```

Comment:
A minor bug fix: don't try to serialize metadata when metadata is not there. serialize it only when it exists.